gpio-ch341.c

```diff
@@ -378,6 +378,15 @@ static int ch341_gpio_probe(struct platform_device *pdev)
 	gpio->set_multiple = ch341_gpio_set_multiple;
 	gpio->base = -1;
 	gpio->ngpio = CH341_GPIO_NUM_PINS;
+	gpio->names = (char const * const []){
+		"D0", "D1", "D2", "D3", "D4", "D5", "D6", "D7",
+		// names are from "parallel" mode, seems to be what datasheets us
+		"ERR#", "PEMP", "INT#", "SLCT", "RST#", "WAIT#", "DS#", "AS#",
+		// names are from "print port" mode, seems to be what datasheets us
+		// "ERR#", "PEMP", "ACK#", "SLCT", "INI#", "BUSY", "AFD#", "SIN#",
+
+		// No WR#?
+	};
```

Comment:
Linux doesn't support that when there are 2 or more devices. I had that code a while ago and removed it for that reason.

Reply:
as of 6.8.0 it does support it with a warning.:
```
 * Take the names from gc->names and assign them to their GPIO descriptors.
 * Warn if a name is already used for a GPIO line on a different GPIO chip.
 *
 * Note that:
 *   1. Non-unique names are still accepted,
 *   2. Name collisions within the same GPIO chip are not reported.
```

But yeah.  What other drivers seem to do is append the chip name to it names:
```
			name = devm_kasprintf(gpio->gpio.parent, GFP_KERNEL,
					      "P%s.%02x", port->name, j);
```